pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -3434,11 +3435,30 @@ func TestValidateParamArrayIndex_valid(t *testing.T) {
 			}},
 		},
 		params: []Param{{Name: "second-param", Value: *NewStructuredValues("second-value", "second-value-again")}},
+	}, {
+		name: "validation with alpha gate",
+		original: PipelineSpec{
+			Params: []ParamSpec{
+				{Name: "first-param", Type: ParamTypeArray, Default: NewStructuredValues("default-value", "default-value-again")},
+				{Name: "second-param", Type: ParamTypeString},
+			},
+			Tasks: []PipelineTask{{
+				Params: []Param{
+					{Name: "first-task-first-param", Value: *NewStructuredValues("$(params.first-param[1])")},
+					{Name: "first-task-second-param", Value: *NewStructuredValues("$(params.second-param[0])")},
+					{Name: "first-task-third-param", Value: *NewStructuredValues("static value")},
+				},
+			}},
+		},
+		params:    []Param{{Name: "second-param", Value: *NewStructuredValues("second-value", "second-value-again")}},
+		apiFields: config.AlphaAPIFields,
 	},
 	} {
 		tt := tt // capture range variable
+		if tt.apiFields == config.AlphaAPIFields {
```

Comment:
you can use this helper function: https://github.com/tektoncd/pipeline/blob/main/pkg/apis/config/feature_flags.go#L351

Reply:
Yeah I noticed that function, but it is an unexported function, do you suggest we export it?

Reply:
yeah I think it would be fine to export!